src/imputation/MoR.py

```diff
@@ -19,17 +21,26 @@ def run_mor(df, backdata, impute_vars):
         df (pd.DataFrame): Processed full responses DataFrame
         backdata (pd.DataFrame): One period of backdata.
         impute_vars ([string]): List of variables to impute.
+        lf_target_vars ([string]): List of long form target vars.
 
     Returns:
-        pd.DataFrame: df with MoR applied
+        pd.DataFrame: df with MoR applied.
+        pd.DataFrame: QA DataFrame showing how imputation links are calculated.
     """
     to_impute_df, remainder_df, backdata = mor_preprocessing(df, backdata)
 
     # Carry forwards method
     carried_forwards_df = carry_forwards(to_impute_df, backdata, impute_vars)
 
-    # TODO Remove the `XXX_prev` columns (left in for QA)
-    return pd.concat([remainder_df, carried_forwards_df]).reset_index(drop=True)
+    gr_df = calculate_growth_rates(remainder_df, backdata, lf_target_vars)
+    links_df = calculate_links(gr_df, lf_target_vars, config)
+
+    carried_forwards_df = apply_links(
+        carried_forwards_df, links_df, lf_target_vars, config
+    )
+    # Calculate totals as with TMI
+    carried_forwards_df = calculate_totals(carried_forwards_df)
```

Comment:
@AnneONS Have added this line to calculate the totals, but didn't know if we should just call the function once in `imputation_main`?

Reply:
it was the original plan to do that... but if so, we'd need to do it separately for long forms, where we need to calculate emp_total and headcount_total, and for shortforms, where those 2 are actually target vars, and different totals are needed.

Reply:
Makes sense, are you happy for me to merge?